lib/trie/proof/generate.go

```diff
@@ -22,7 +23,7 @@ var (
 // Database defines a key value Get method used
 // for proof generation.
 type Database interface {
-	Get(key []byte) (value []byte, err error)
+	db.DBGetter
 }
```

Comment:
Since this interface is just a wrapper for `db.DBGetter` I would suggest to remove it and use `db.DBGetter` directly